### PR TITLE
Adds the interface to complete a timed out job

### LINKED ISSUE #130: Add UI for completeTimedOutJob
When a payment request times out, the engineer should be able to force completion and payment

### PR DESCRIPTION
**Pull request by deweller**
Closes #130 

image

image

image

Note that this is a stacked PR.

### CODE REVIEW

File: frontend/src/components/single-contract/TimeoutJobForm.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const history = useHistory();
+
+  const { contracts } = useSmartContracts();
+
+  // get the job timeout
+  const [jobTimeoutSeconds, timeoutLoading] =
+    useSmartContractCallback<BigNumber>(() => {
+      return contracts.Job.COMPLETED_TIMEOUT_SECONDS();
+    });
+  console.log('rendering TimeoutJobForm');
```
Comment (reisr3) [inline frontend/src/components/single-contract/TimeoutJobForm.tsx:21]: We should add the lint rule for console logs. Mind removing this?

File: frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState, useCallback } from 'react';
+
+type CallContractCallback = () => Promise<any>;
+
+export const useSmartContractCallback = <ResultType>(
```
Comment (reisr3) [inline frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts:5]: I don't really like this hook. It's hard to tell what it's doing. Can't we just add this to `job` and do

```
const {job} = useJob();
if job.timeout > date.now() etc...
```

IMHO - this is stuff that should be abstracted behind our existing hooks so its very easy to use at component level.

File: frontend/src/components/single-contract/TimeoutJobForm.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              disabled={!jobIsTimedOut || showTimeoutJobModal}
+              type="submit"
+              className={
+                'focus:outline-none inline-flex items-center rounded-md border border-transparent px-4 py-2 shadow-sm focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2' +
+                (!jobIsTimedOut || showTimeoutJobModal
+                  ? ' bg-indigo-100 hover:bg-indigo-100'
+                  : ' bg-indigo-600 hover:bg-indigo-700')
+              }
+            >
+              Receive Timed Out Job Payment
```
Comment (reisr3) [inline frontend/src/components/single-contract/TimeoutJobForm.tsx:66]: Can we change this to `Receive Payment` and maybe make it green?

**Review by reisr3 — CHANGES_REQUESTED** [review]
I'm cool with it if this is the easiest way to do it right now - but shouldn't we just pay the engineer?

I figured we would just send the money to their wallet upon timeout, without them having to come back and do anything. That way the contract is not in a weird limbo.

File: frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts (hunk 2 above)
Comment (deweller) [reply frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts:5]: It is not a call on a specific job.  It is a call to the Job smart contract.

I was trying to make a general purpose hook.  But I think what you are looking for is a more specific hook like:

```typescript
const [jobTimeoutSeconds, timeoutLoading] = useJobContractVariable('COMPLETED_TIMEOUT_SECONDS');
```

Comment (deweller) [reply frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts]: I created a `useJobContractVariable` hook.  Is there a better way?

Comment (deweller) [reply frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts]: I refactored this to `useSmartContractCall`.  I think this is the cleanest way to do this.

See https://github.com/engineer-dao/platform/pull/136/commits/4f21a31020e4e88ffa29792b843ed1dea985e02f#diff-1d48d830e080e7d6c77b002a29f0025f0f8b86a4da4a00f795782439ec9d1cd2R16-R19

**Review by reisr3 — APPROVED** [review]
🚀

**Pull request merged**